consensus/enclave/impl/src/lib.rs

```diff
@@ -177,20 +180,25 @@ impl ConsensusEnclave for SgxConsensusEnclave {
         peer_self_id: &ResponderId,
         client_self_id: &ResponderId,
         sealed_key: &Option<SealedBlockSigningKey>,
-        minimum_fee: Option<u64>,
+        minimum_fees: Option<BTreeMap<TokenId, u64>>,
     ) -> Result<(SealedBlockSigningKey, Vec<String>)> {
-        // Inject the fee into the peer ResponderId
-        let peer_self_id = if let Some(fee) = minimum_fee {
-            let str = format!("{}-{}", &peer_self_id.0, fee);
-            ResponderId(str)
-        } else {
-            peer_self_id.clone()
-        };
+        // If we are provided with the optional fee map, we must have >0 MOB minimum
+        // fee.
+        if let Some(ref minimum_fees) = minimum_fees {
+            if minimum_fees.get(&TokenId::MOB).cloned().unwrap_or_default() == 0 {
+                return Err(Error::InvalidFeeConfig);
+            }
+        }
 
+        // Inject the fee into the peer ResponderId
+        let peer_self_id = ResponderId(format!(
+            "{}-{}",
+            &peer_self_id.0,
+            self.fee_map.get_digest_str()
+        ));
```

Comment:
Is it useful to invert this logic to avoid the string copy?

Reply:
I don't understand what you're referring to - what logic would you invert here?

Reply:
`self.fee_map.responder_id(&peer.self_id.0);`, and then that method takes the lock, formats the string, and returns the result.

Reply:
Ah nice, I like that. Will implement.